spark/src/main/kotlin/com/adevinta/spark/components/toggles/RadioButton.kt

```diff
@@ -158,50 +216,101 @@ public fun RadioButtonLabelled(
+ * @param contentSide The side where we want to show the label, default to [ContentSide.End].
+ * @param content The content displayed after the radio button, usually a Text composable shown at the end.
+ */
 @Composable
-internal fun AllStatesRadioButtonPreview() {
-    PreviewTheme {
-        ToggleIntent.values().forEach { intent ->
-            Row {
-                RadioButton(enabled = true, selected = true, onClick = {}, intent = intent)
-                RadioButton(enabled = false, selected = true, onClick = {}, intent = intent)
-                RadioButton(enabled = true, selected = false, onClick = {}, intent = intent)
-                RadioButton(enabled = false, selected = false, onClick = {}, intent = intent)
-            }
-        }
-    }
+public fun RadioButtonLabelled(
+    selected: Boolean,
+    onClick: (() -> Unit)?,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    error: Boolean = false,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    contentSide: ContentSide = ContentSide.End,
+    content: @Composable RowScope.() -> Unit,
+) {
+    SparkToggleLabelledContainer(
+        state = ToggleableState(selected),
+        toggle = {
+            RadioButton(
+                modifier = it.minimumTouchTargetSize(),
+                selected = selected,
+                onClick = null,
+                enabled = enabled,
+                error = error,
+                interactionSource = interactionSource,
+            )
+        },
+        role = Role.RadioButton,
+        onClick = onClick,
+        modifier = modifier,
+        enabled = enabled,
+        contentSide = contentSide,
+        content = content,
+    )
 }
 
 @Preview(
     group = "Toggles",
     name = "RadioButtonLabelled",
+    device = Devices.FOLDABLE,
 )
 @Composable
-internal fun AllStatesRadioButtonLabelledPreview() {
+private fun PreviewAllStatesRadioButtonLabelled() {
     PreviewTheme {
-        RadioButtonLabelled(
-            enabled = true,
-            selected = true,
-            onClick = {},
-        ) { Text("RadioButton On") }
-        RadioButtonLabelled(
-            enabled = false,
-            selected = true,
-            onClick = {},
-        ) { Text("RadioButton On") }
-        RadioButtonLabelled(
-            enabled = true,
-            selected = false,
-            onClick = {},
-        ) { Text("RadioButton Off") }
-        RadioButtonLabelled(
-            enabled = false,
-            selected = false,
-            onClick = {},
-        ) { Text("RadioButton Off") }
+        RadioButtonStates(isError = false)
+        RadioButtonStates(isError = true)
     }
 }
+
+@Composable
+private fun ColumnScope.RadioButtonStates(isError: Boolean) {
+    RadioButtonLabelled(
+        enabled = true,
+        selected = true,
+        onClick = {},
+        error = isError,
+    ) { Text("RadioButton On") }
```

Comment:
Same thing here. They are all `RadioButton`s.

Reply:
I changed them by possible options for toggles